test/utilities/ci.jl

```diff
@@ -0,0 +1,47 @@
+@testset "auto_detect_ci_service" begin
+    @testset "env var present" begin
+        withenv(
+            "GITHUB_REPOSITORY" => "true"
+        ) do
+            expected = CompatHelper.GitHubActions()
+
+            @test CompatHelper.auto_detect_ci_service() == expected
+        end
+    end
+
+    @testset "env var dne" begin
+        withenv(
+            "GITHUB_REPOSITORY" => "foobar"
+        ) do
+            delete!(ENV, "GITHUB_REPOSITORY")
+            @test_throws CompatHelper.UnableToDetectCIService CompatHelper.auto_detect_ci_service()
+        end
+    end
+end
+
+@testset "set_git_identity" begin
+    function read_git_config()
+        try
+            name = strip(read(`git config user.name`, String))
+            email = strip(read(`git config user.email`, String))
+
+            return name, email
+        catch
+            return "", ""
```

Comment:
maybe put a note here saying that this might happen if the commands return nothing
Might be worth putting the user.name and user.email requests in different try catches as well? Not too sure when one might work but the other wont, but it could be possible.